zppy/e3sm_diags.py

```diff
@@ -64,6 +64,29 @@ def e3sm_diags(config, scriptDir):  # noqa: C901
                     c["ref_year1"],
                     c["ref_year2"],
                 )
+                reference_data_path = (
+                    c["reference_data_path"].split("/post")[0] + "/post"
+                )
+                if c["reference_data_path_climo_diurnal"] == "":
+                    c[
+                        "reference_data_path_climo_diurnal"
+                    ] = f"{reference_data_path}/atm/{c['grid']}/clim_diurnal_8xdaily"
+                if c["reference_data_path_tc"] == "":
+                    c[
+                        "reference_data_path_tc"
+                    ] = f"{reference_data_path}/atm/tc-analysis_{c['ref_year1']}_{c['ref_year2']}"
+                if c["reference_data_path_ts"] == "":
+                    c[
+                        "reference_data_path_ts"
+                    ] = f"{reference_data_path}/atm/{c['grid']}/ts/monthly"
+                if c["reference_data_path_ts_rof"] == "":
+                    c[
+                        "reference_data_path_ts_rof"
+                    ] = f"{reference_data_path}/rof/native/ts/monthly"
+                if c["gauges_path"] == "":
+                    c[
+                        "gauges_path"
+                    ] = "/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/obs/time-series/GSIM/GSIM_catchment_characteristics_all_1km2.csv"  # Define with Mache?
```

Comment:
Should we leave this hard-coded for now? I'm not sure what would be involved to use Mache here. Wouldn't Mache have to define a `gauges_path` for each machine?

I think we'd also need to define Mache as a dependency in the `conda` environment files too.

Reply:
hmmm, It is a little confusing to have to point to a obs path for `model vs model` run. I think the cleaner way is to include the gauge locations as a separate file, which should install with e3sm_diags. Before having the separation implement, I thank Mache can be used here to determine the obs path per machines.

Reply:
+1 to use Mache for all the necessary paths in E3SM Diags

Reply:
@forsyth2, the easiest might be for us to have a short chat.

Just add `mache >=1.3.2` as a requirement for `zppy` and you'll have it.  It's on conda-forge: https://github.com/conda-forge/mache-feedstock

There's no need to do a development environment for `mache` unless you need to add new functionality.  So far, I don't see any reason you would.

Then, you would get a `MachineInfo` object like described in the readme: https://github.com/E3SM-Project/mache/blob/main/README.rst.  On compute nodes, you need some way to know what machine you're on.  But that can be automatic for `zppy` because it always runs on login nodes, which `mache` will detect automatically.

I believe that E3SM Diags is now storing this data in the "sanctioned" place on each machine, within the shared diagnostics directory.  @chengzhuzhang, please correct me if that's not yet the case.  If so, you can get the diagnostics directory on a given machine most easily with:
``` python
from mache import MachineInfo

...
machine_info = MachineInfo()
diags_base = machine_info.config.get('diagnostics', 'base_path')
```
You can see what other config options are available by looking at an example machine:
https://github.com/E3SM-Project/mache/blob/main/mache/machines/chrysalis.cfg

If E3SM Diags is storing these files in its own place outside of `diags_base`, we're going to have a harder time and we need to discuss this.  But I could (reluctantly) add something new to `mache` specifically for E3SM Diags if that's the best way forward.

Reply:
Just to clarify, the "sanctioned" location for diagnostic data is:
```
/lcrc/group/e3sm/public_html/diagnostics/
```
if it can be available to the public or:
```
/lcrc/group/e3sm/diagnostics_private/
```
if it's supposed to be available only to E3SM team members.  Then, the `mache` sync functionality (https://github.com/E3SM-Project/mache#syncing-diagnostics) that I just added can be used to (manually) sync this data to `diags_base` on each machine.

Reply:
Yes, the same streamflow data is here: /lcrc/group/e3sm/public_html/diagnostics/observations/Atm/time-series/GSIM/GSIM_catchment_characteristics_all_1km2.csv

I haven't used mache to populate the same /lcrc/group/e3sm/public_html/diagnostics/ directory on other machines. This is a good opportunity to unify the location . [so we don't need to maintain separate e3sm_diags data in another path on different machines.]

Reply:
@chengzhuzhang, once LCRC is back up, I can run the sync script on all machines (except Acme1, as usual).

Reply:
Thank you @xylar ! i will take care of acme1 :)

Reply:
@chengzhuzhang, I just synced on Anvil/Chrysalis, Andes, Compy, Cooley and Cori.  In all cases, there was nothing to sync.  I think I had told you not to worry about Acme1 until the next `mache` release and I think that advice still holds for now unless you want to test this PR there.